encodings/internal.js

```diff
@@ -196,3 +198,30 @@ InternalDecoderCesu8.prototype.end = function() {
         res += this.defaultCharUnicode;
     return res;
 }
+
+//------------------------------------------------------------------------------
+// check the chunk boundaries for surrogate pair
+
+function InternalEncoderUtf8(options, codec) {
+    this.surrogateLeftStr = '';
+}
+
+InternalEncoderUtf8.prototype.write = function (str) {
+    var lastr = str[str.length - 1]
+    if ('\uD800' < lastr && lastr <= '\uDBFF') {
+        this.surrogateLeftStr += str
```

Comment:
this has the potential of accumulating unlimited data in this.surrogateLeftStr as we stream.. Can we encode and return everything but the last character instead?